db/migrate/20170501201633_add_canceled_to_order_detail.rb

```diff
@@ -4,10 +4,11 @@ def up
     add_column :order_details, :canceled_by, :integer
     add_column :order_details, :canceled_reason, :string
 
-    Reservation.where.not(order_detail_id: nil).find_each do |reservation|
-      reservation.order_detail.update_attributes(canceled_at: reservation.canceled_at,
+    Reservation.where.not(order_detail_id: nil, canceled_at: nil).find_each do |reservation|
+      reservation.order_detail.assign_attributes(canceled_at: reservation.canceled_at,
```

Comment:
Not sure how big a deal this is, but maybe define a dummy `Reservation` class at the top of the file (as in http://railsguides.net/change-data-in-migrations-like-a-boss/) -- I normally prefer data changes to be outside migrations, but I understand that might be logistically complicated.